File: C#/Prefix-Sum-And-Sufix-Sum.cs
// C# Program for Implementing
// prefix sum arrayclass
using System;

class GFG {
	// Fills prefix sum array
	static void fillPrefixSum(int[] arr, int n,
							int[] prefixSum)
	{
		prefixSum[0] = arr[0];

		// Adding present element
		// with previous element
		for (int i = 1; i < n; ++i)
			prefixSum[i] = prefixSum[i - 1] + arr[i];
	}

	// Driver code
	public static void Main()
	{
		int[] arr = { 10, 4, 16, 20 };
		int n = arr.Length;
		int[] prefixSum = new int[n];

		fillPrefixSum(arr, n, prefixSum);

		for (int i = 0; i < n; i++)
			Console.Write(prefixSum[i] + " ");
		Console.Write("");
	}
}

// This Code is Contributed by nitin mittal

File: Php/Prefix-Sum-And-Sufix-Sum.php
<?php
// PHP program for
// Implementing prefix
// sum array

// Fills prefix sum array
function fillPrefixSum($arr,
					$n)
{
	$prefixSum = array();
	$prefixSum[0] = $arr[0];

	// Adding present element
	// with previous element
	for ($i = 1; $i < $n; $i++)
		$prefixSum[$i] = $prefixSum[$i - 1] +
									$arr[$i];
		
	for ($i = 0; $i < $n; $i++)
		echo $prefixSum[$i] . " ";
}

// Driver Code
$arr = array(10, 4, 16, 20);
$n = count($arr);

fillPrefixSum($arr, $n);

// This code is contributed
// by Sam007
?>
